src/pyird/image/wavcal.py

```diff
@@ -1,11 +1,85 @@
+from pyird.io.read_linelist import read_linelist
 from astropy.io import fits
 import numpy as np
 from scipy.signal import medfilt
+from scipy.optimize import leastsq
 import pandas as pd
 
+import matplotlib.pyplot as plt
+
+def pdat2matrix(pdat):
+    data = np.zeros((l-j,2048))
+    m0 = j+1
+    for i in range(len(pdat)):
+        order, pixel, wav = pdat.iloc[i].values
+        order = order - m0
+        data[order,pixel] = wav
+    return data
+
+def fitfunc(XY,Ni,Nx,*params):
+    pix = XY[0][0]
+    m = XY[1][:,0]
+    minv = 1/XY[1]
+    p=[]
```

Comment:
```python
p = np.array(params).reshape((Ni,Nx))
```
is enough?

Reply:
Thank you for pointed it out. I have changed it.